### PR TITLE
kv: migrate test-infra to testify

### LINKED ISSUE #26100: migrate test-infra to testify for `kv` pkg
This package contains tests for leak. Replace manually leak scripts with [go.uber.org/goleak](https://github.com/uber-go/goleak) as in https://github.com/tikv/client-go/pull/172

### LINKED ISSUE #26107: Adopt go.uber.org/goleak for leak tests

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number:

- closes #26100
- closes #26107 

### What is changed and how it works?

How it Works:

Line by line migrate matchers. Also replace manually leak scripts with [go.uber.org/goleak](https://github.com/uber-go/goleak) as in https://github.com/tikv/client-go/pull/172

### Release note 

- No release note

### CODE REVIEW

File: kv/mock_test.go [hunk 1]
```diff
@@ -15,90 +15,75 @@ package kv
...
-	c.Assert(status, IsNil)
-	c.Assert(err, IsNil)
+	assert.Nil(t, status)
+	assert.Nil(t, err)
 
 	err = storage.Close()
-	c.Check(err, IsNil)
-}
-
-func (s testMockSuite) TestIsPoint(c *C) {
```
Comment (tisonkun) [inline kv/mock_test.go:89]: Case has been covered by `key_test.go`

**Review by disksing — APPROVED** [review]

File: kv/utils_test.go [hunk 2]
```diff
@@ -18,39 +18,76 @@ import (
 	"strconv"
 	"testing"
 
-	. "github.com/pingcap/check"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestT(t *testing.T) {
-	CustomVerboseFlag = true
-	TestingT(t)
```
Comment (tiancaiamao) [inline kv/utils_test.go:26]: Is all the tests translated in this PR?
After removing `TestingT`,  the testsuites will not run...

**Review by tiancaiamao — APPROVED** [review]

File: kv/utils_test.go (hunk 2 above)
Comment (tisonkun) [reply kv/utils_test.go:26]: Yes. The suite itself is removed as migrated.

**Comment by tisonkun** [comment]
/assign @tiancaiamao @disksing 

Thanks for your review! May you help on merging this PR?

**Pull request merged**